### DISCUSSION TITLE
Custom Control Update from 0.10.18 -> 11

### DISCUSSION THREAD

**Discussion by jamsoft** [AvaloniaUI/Avalonia#15053] [General]
Hi All,

I've been trying to get my head around all the changes between these two versions in relation to a custom control I'd written. I'm failing at the moment. A lot of the things I was using/implemented have gone internal and I'm a bit confused about how to proceed.

I'm having trouble with this control:

https://github.com/jamsoft/JamSoft.AvaloniaUI.Dialogs/tree/master/src/JamSoft.AvaloniaUI.Dialogs/Controls

Are there any docs online related to updating custom controls that are worth reading?

**Comment by kekekeks** [comment]
https://docs.avaloniaui.net/docs/stay-up-to-date/upgrade-from-0.10#itemscontrol
